server/src/main/java/org/candlepin/async/tasks/EntitleByProductsJob.java

```diff
@@ -0,0 +1,183 @@
+        entitler = e;
+        consumerCurator = c;
+    }
+
+    @Override
+    public Object execute(final JobExecutionContext ctx) throws JobExecutionException {
+        try {
+            final JobArguments arguments = ctx.getJobArguments();
+            final String consumerUuid = arguments.getAsString(CONSUMER_UUID_KEY);
+            final Date entitleDate = arguments.getAs(ENTITLE_DATE_KEY, Date.class);
+            final String[] prodIds = arguments.getAs(PROD_IDS_KEY, String[].class);
+            final String[] fromPools = arguments.getAs(FROM_POOLS_KEY, String[].class);
+
+            final List<Entitlement> ents = entitler.bindByProducts(
+                prodIds, consumerUuid, entitleDate, Arrays.asList(fromPools));
+            entitler.sendEvents(ents);
+            return "Entitlements created for owner";
+        }
+        // Catch any exception that is fired and re-throw as a JobExecutionException
+        // so that the job will be properly cleaned up on failure.
+        catch (Exception e) {
+            log.error("EntitlerJob encountered a problem.", e);
+            throw new JobExecutionException(e.getMessage(), e, false);
+        }
+    }
+
+    public static EntitleByProductsJobConfig createConfig() {
+        return new EntitleByProductsJobConfig();
+    }
+
+    /**
+     * Job configuration object for the entitle by products job
+     */
+    public static class EntitleByProductsJobConfig extends JobConfig {
+
+        public EntitleByProductsJobConfig() {
+            this.setJobKey(JOB_KEY)
+                .setJobName(JOB_NAME)
+                .addConstraint(JobConstraints.uniqueByArgument(OWNER_KEY));
+        }
+
+        public EntitleByProductsJobConfig setOwner(final Owner owner) {
+            if (owner == null) {
+                throw new IllegalArgumentException("Owner is null");
+            }
+
+            this.setJobMetadata(OWNER_KEY, owner.getKey())
+                .setJobArgument(OWNER_KEY, owner.getKey())
+                .setLogLevel(owner.getLogLevel());
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setConsumer(final Consumer consumer) {
+            if (consumer == null) {
+                throw new IllegalArgumentException("Consumer is null");
+            }
+            this.setJobArgument(CONSUMER_UUID_KEY, consumer.getUuid());
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setProductIds(final String[] prodIds) {
+            if (prodIds == null) {
+                throw new IllegalArgumentException("Product ids is null");
+            }
+            this.setJobArgument(PROD_IDS_KEY, prodIds);
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setEntitleDate(final Date entitleDate) {
+            this.setJobArgument(ENTITLE_DATE_KEY, entitleDate);
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setFromPools(final Collection<String> fromPools) {
```

Comment:
Careful on the method name here, this is just verbiage from the original author's variable names being carried forward. This should just be `setPools`.

Reply:
Done.